src/NServiceBus.Core.Tests/API/APIApprovals.ApproveNServiceBus.approved.txt

```diff
@@ -1872,6 +1879,21 @@ namespace NServiceBus.Persistence
     }
     public interface SynchronizedStorageSession { }
 }
+namespace NServiceBus.Persistence.Legacy
+{
+    [System.ObsoleteAttribute("The classes under the NServiceBus.Persistence.Legacy namespace have been moved. R" +
+        "eference the new NServiceBus.Transport.Msmq package and use NServiceBus.MsmqPers" +
+        "istence instead. Will be removed in version 8.0.0.", true)]
+    public class MsmqPersistence : NServiceBus.Persistence.PersistenceDefinition
```

Comment:
is this obsolete message worth having the same class available in two namespaces, always requiring the user the make a conscious choice of the used namespace?

Reply:
I'm happy to go with whatever @Particular/nservicebus-maintainers agree upon. Let me know when you

Reply:
Why would we not obsolete in one major and remove in next in this case?

Reply:
> Why would we not obsolete in one major and remove in next in this case?

in general, obsoleting APIs is preferable as we have a way to guide users to the new API's. In cases where the whole API is moved more or less 1:1 into a different package this is a bit trickier as autocomplete/R# will not know which class you meant to use and will show you both options. There are three options I think:
* go with the obsoletes regardless.
* use a different public API on MSMQ using different types so you don't have "ambiguous" types
* remove without obsoletes and leave users using MSMQ only with the upgrade guide to figure out what to do.

Reply:
Yeah, the options are clear. The msmq extraction is probably not breaking any new ground here, though. What has been done in these cases before?

Reply:
not sure we had another package with the same situation. MessagePropertyEncryption introduced a new API instead. I thought MSMQ was the first one using a 1:1 extraction approach (and that's why there were no obsoletes in the first place).